src/org/labkey/test/LabKeySiteWrapper.java

```diff
@@ -720,6 +722,29 @@ else if (getDriver().getTitle().startsWith("Sign In"))
                     // Note: leave the self-report setting unchanged
                     customizeSitePage.save();
                 }
+
+                //Waiting for search service to boot up
+                Connection connection = createDefaultConnection();
+                SimpleGetCommand command = new SimpleGetCommand("search", "json");
+                command.setParameters(Map.of("q", "pinging to check server is started", "scope", "All"));
+                long searchEndTime = System.currentTimeMillis() + 180000; // Wait for maximum of 3 mins.
+                CommandResponse searchResponse = null;
+                do
+                {
+                    try
+                    {
+                        searchResponse = command.execute(connection, "/");
+                    }
+                    catch (IOException e)
+                    {
+                        throw new RuntimeException(e);
+                    }
+                    catch (CommandException e)
+                    {
+                        sleep(500); // Retry and to poll the re-request
+                    }
+                }
+                while ((searchResponse != null) && (searchResponse.getStatusCode() != 200) && (System.currentTimeMillis() < searchEndTime));
```

Comment:
If the command throws an exception, `searchResponse` will be null and this won't loop.